### ISSUE TITLE
Remove all Python 2.x items

### ISSUE THREAD

**Issue by eteq** [astropy/astropy#6234]
This issue is a placeholder "master issue" for one of the main tasks of 3.0: removing all Python 2.x compatibility bits to make 3.0 cleaner given that 2.x will no longer be supported.

Note that the current plan is to wait on merging anything in this direction for about halfway through the 3.0 development cycle.  That gives a window for some bugfixes in 2.x without having to deal with backport hell.

(It's *possible* there's another issue to track this, although I don't think so.  @astrofrog, do you know?)

**Comment by bsipocz** [comment]
@eteq - I think #4454 is about this about this, but a bit more vague as was opened before the timeline was decided. I'm going ahead and close it.

**Comment by astrofrog** [comment]
We should brainstorm on what things need to be done so that we don't go over the code multiple times:

* Remove ``__future__`` imports
* Make sure we remove the ``unicode_literals`` text fixture
* Search for and remove any 'six' imports and in some cases replace by the appropriate Python 3 import.
* Search for and remove any ``if six.PY2`` blocks and also search for ``sys.version_info``
* Update CI configuration to not run tests on Python 2
* Search for comments that mention Python 2 and 3 to see if anything can be updated

One way to find code that might be Python 2-specific is to run the coverage tests for Python 2 and 3 and figure out which lines were covered in 2 and not 3. This could be done in a scripted way.

Anything else I'm missing?

**Comment by bsipocz** [comment]
I've edited your list, but we also may want to start with merging off the already opened PRs, and go through the list of issues labelled either ``Python3`` or ``Python2`` to see whether they are relevant.

**Comment by astrofrog** [comment]
Yeah a lot of PRs are going to have conflicts after this...

**Comment by bsipocz** [comment]
Oh, I meant the PRs that are related to this, e.g. #5978 and alike.

**Comment by bsipocz** [comment]
But if you suggest a big triaging sprint before that, I'm all in that, too.

**Comment by bsipocz** [comment]
@astrofrog - This is loosely connected, but for the new release, we may want to also go over of the warnings in the module docs, and remove warnings when the feature is mature enough (e.g. https://github.com/astropy/astropy/pull/6538#issuecomment-328889306 points to one in the table indexing, but that was new in 1.1)

**Comment by bsipocz** [comment]
#6655 should probably address the last one from @astrofrog's list above, so maybe we can close this issue once that is merged?

**Comment by MSeifert04** [comment]
#6678 is also related to this. But then I can't think of anything else where we could have python2 related stuff.

**Comment by bsipocz** [comment]
I'm closing this now as we tried to do everything from the list above. It's highly probable that there are still a few things out there that are overlooked, but no need to keep the placeholder open any more.
